cloud/azure/deploy/policy.go

```diff
@@ -144,6 +144,9 @@ func (p *NitricAzurePulumiProvider) scopeFromResource(resource *deploymentspb.Re
 func (p *NitricAzurePulumiProvider) Policy(ctx *pulumi.Context, parent pulumi.Resource, name string, policy *deploymentspb.Policy) error {
 	opts := []pulumi.ResourceOption{pulumi.Parent(parent)}
 
+	// used to ensure we don't assign the same role multiple times
+	assignedRoles := make(map[string]bool)
+
```

Comment:
should be able to remove this with unique pulumi resource naming (including target resource).